build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -298,50 +300,69 @@ export class CheckPolicy extends BaseCommand<typeof CheckPolicy> {
 		}
 	}
 
-	private handleLine(line: string, commandContext: CheckPolicyCommandContext): void {
+	/**
+	 * Given a string that represents a path to a file in the repo, determines if the file should be checked, and if so,
+	 * routes the file to the appropriate handlers.
+	 */
+	private async checkOrExcludeFile(
+		inputPath: string,
+		commandContext: CheckPolicyCommandContext,
+	): Promise<void> {
```

Comment:
Do you have plans for this one to eventually do async stuff too? I could see `fs.exists()` over `fs.existsSync()`. It feels separate enough from the flow of executing the handlers that I'd lean not making it async in this PR, but won't block on it. I love the renames though.

Reply:
Now that I am properly awaiting the .map results, it needs to be async.

Reply:
Scratch that, I was wrong. Fixing.

Reply:
This is called in a different `map()` than the execution of the handlers; that one could stay completely sync for now. But if we move ahead with making this async, yeah, both `map()`s should get the promises awaited.

Reply:
checkOrExcludeFile calls routeToHandlers, which is async.

Reply:
Oh you're right, I had missed that :).